packages/request-client.js/src/api/request.ts

```diff
@@ -120,7 +130,11 @@ export default class Request {
       requestId: this.requestId,
     };
 
-    await this.requestLogic.cancelRequest(parameters, signerIdentity, true);
+    try {
+      await this.requestLogic.cancelRequest(parameters, signerIdentity, true);
+    } catch (error) {
+      throw new Error(`${error.message} (maybe transactions are still pending)`);
```

Comment:
we could actually verify the state before trying these actions, and fail with that message only if relevant.

Reply:
Finnaly we check also the pending state